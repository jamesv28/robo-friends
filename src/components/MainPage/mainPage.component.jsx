import React, { Component } from 'react';

import CardList from '../CardList/cardlist.component';
import Header from '../Header/header.component';
import Scroll from '../../containers/Scroll/Scroll';
import ErrorBoundary from '../Errorboundary/errorboundary.component';

import './mainPage.style.css';

class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
      return this.props.robots.filter(robot => {
        return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
      })
  };

  render() {

    const {onSearchChange, robots, isPending} = this.props;

    return !isPending ?
    (
      <>
        <Header searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundary>
            <CardList robots={this.filteredRobots(robots)}/>
          </ErrorBoundary>
        </Scroll>
      </>
     ) :
     (
      <header>
        <h1 className="tc" aria-busy="true" role="alert">Loading...</h1>
      </header>
     ) 

  }
}

export default MainPage;

