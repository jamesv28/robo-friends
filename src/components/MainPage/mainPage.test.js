import React from 'react';
import MainPage from './mainPage.component';
import {shallow} from 'enzyme';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
})

it('renders main page without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filters robots correctly', () => {
    expect(wrapper.instance().filteredRobots([])).toEqual([]);
})

it('filters the correct user', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [
            {
                id: 3,
                name: 'hohn',
                email: 'john@gmail.com'
            }
        ],
        searchField: 'a',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filteredRobots()).toEqual([])
})