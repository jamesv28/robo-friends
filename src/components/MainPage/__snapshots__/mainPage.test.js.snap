// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main page without crashing 1`] = `ShallowWrapper {}`;
